### DISCUSSION TITLE
useController not focusing input on error

### DISCUSSION THREAD

**Discussion by hannarimon** [react-hook-form/react-hook-form#9859] [Q&A]
Hello!

I am using FormProvider and giving Datepicker name and control through props.

![image](https://user-images.githubusercontent.com/70326284/216351855-edc2a697-6c14-4cb5-addb-7048fe2589c0.png)

In the Datepicker I use useController and spread the field to the input.

![image](https://user-images.githubusercontent.com/70326284/216353400-165922c3-681d-4b75-b2e8-7f229a5cda2b.png)

![image](https://user-images.githubusercontent.com/70326284/216353117-5d3ac4ce-6972-41a1-9b64-54697025dca3.png)

Using console.log(formState.errors) shows that the ref is not being applied correctly. It should have the actual input as the ref.

![image](https://user-images.githubusercontent.com/70326284/216355254-b9bd0f70-af2b-4770-a616-360a49bf9ef0.png)

I found this solution on another discussion but with no further explanation. I know that useController should automatically connect the ref when spreading the field. I have been googling for over two weeks and would love some help to find the "proper" solution. Thank you!

[The solution I am using for now.](https://github.com/react-hook-form/react-hook-form/discussions/5435)

**Comment by Moshyfawn** [comment]
That's the thing: `useController` should **not** automatically connect the `ref`.

There can be different causes of why the `ref` is not registered with RHF:
- You override the ref props after spreading the field props
- The component / input API `ref` prop is with a different name _(`inputRef`)_
- The `ref` is not the expected object on the element

From your screenshots, I can't see why RHF would get a plain object for the `ref`. Please, provide a minimal reproduction sandbox, so we can debug further

**Reply by hannarimon** [reply]
Hey! Sorry for the slow answer. I was creating a CodeSandbox and tested it out. Turns out, spreading the field on the input works. I also found what was causing it to not work.
![image](https://user-images.githubusercontent.com/70326284/216582867-bc43013d-8550-4716-b407-b6bb14e3c58b.png)
I don't understand what shouldUnregister does. I don't understand how it is impacting the form and refs for useController. I have been reading up on it but to no avail. I would love an explanation on it!
